solver/src/driver.rs

```diff
@@ -204,25 +175,31 @@ impl Driver {
         tracing::debug!("solving with gas price of {}", gas_price);
 
         let settlements = self
-            .run_solvers(liquidity, gas_price, &estimated_prices)
-            .await;
-        for settlements in settlements.iter() {
-            for settlement in settlements.settlements.iter() {
-                tracing::debug!(
-                    "solver {} found solution with objective value {}: {:?}",
-                    self.solver[settlements.index].name(),
-                    settlement.objective_value,
-                    settlement.settlement,
-                );
-            }
-        }
+            .run_solvers(liquidity, gas_price)
+            .await
+            .filter_map(solver_settlements::filter_bad_settlements)
+            .inspect(|(name, settlements)| {
+                for settlement in settlements.iter() {
+                    tracing::debug!("solver {} found solution:\n {:?}", name, settlement,);
```

Comment:
ubernit: trailing comma
```suggestion
                    tracing::debug!("solver {} found solution:\n {:?}", name, settlement);
```